execute/aggregate.go

```diff
@@ -36,176 +162,157 @@ func (c AggregateConfig) Copy() AggregateConfig {
 	return nc
 }
 
-func (c *AggregateConfig) ReadArgs(args flux.Arguments) error {
+func (c *SimpleAggregateConfig) ReadArgs(args flux.Arguments) error {
 	if col, ok, err := args.GetString("column"); err != nil {
 		return err
 	} else if ok {
 		c.Columns = []string{col}
 	} else {
-		c.Columns = DefaultAggregateConfig.Columns
+		c.Columns = DefaultSimpleAggregateConfig.Columns
 	}
 	return nil
 }
 
-func NewAggregateTransformation(d Dataset, c TableBuilderCache, agg Aggregate, config AggregateConfig) *aggregateTransformation {
-	return &aggregateTransformation{
-		d:      d,
-		cache:  c,
+func NewSimpleAggregateTransformation(id DatasetID, agg SimpleAggregate, config SimpleAggregateConfig, mem memory.Allocator) (Transformation, Dataset, error) {
+	tr := &simpleAggregateTransformation{
 		agg:    agg,
 		config: config,
 	}
+	return NewAggregateTransformation(id, tr, mem)
 }
 
-func NewAggregateTransformationAndDataset(id DatasetID, mode AccumulationMode, agg Aggregate, config AggregateConfig, a *memory.Allocator) (*aggregateTransformation, Dataset) {
-	cache := NewTableBuilderCache(a)
-	d := NewDataset(id, mode, cache)
-	return NewAggregateTransformation(d, cache, agg, config), d
+type aggregateState struct {
+	inType flux.ColType
```

Comment:
making it colInType will make it more readable.

Reply:
I find this a bit harder to read personally. I'd like to leave it as is if that's ok from personal preference. I'll add a comment to clarify it though.